src/main/scala/io/iohk/ethereum/network/discovery/Node.scala

```diff
@@ -0,0 +1,62 @@
+package io.iohk.ethereum.network.discovery
+
+import java.net._
+
+import akka.util.ByteString
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.{Failure, Success, Try}
+
+case class Node(id: ByteString, addr: InetSocketAddress) {
+  def toUri: URI = {
+    val host = addr.getHostName
+    val port = addr.getPort
+    new URI(s"enode://${Hex.toHexString(id.toArray[Byte])}@$host:$port")
+  }
+}
+
+object Node {
+  def fromUri(uri: URI): Node = {
+    val nodeId = ByteString(Hex.decode(uri.getUserInfo))
+    Node(nodeId, new InetSocketAddress(uri.getHost, uri.getPort))
+  }
+}
+
+object NodeParser {
+  val NodeScheme = "enode"
+  val NodeIdSize = 64
+
+  /**
+    * Parse a node string, for it to be valid it should have the format:
+    * "enode://[128 char (64bytes) hex string]@[IPv4 address]:[port]"
+    *
+    * @param node to be parsed
+    * @return the parsed node, or the error detected during parsing
+    */
+  def parseNode(node: String): Try[Node] = {
```

Comment:
Should 't we accumulate errors during parsing (for one node there can be 3 of them), and at the end inform a user about all errors in the node, not to force them to correct them one by one? I would imagine return signature would be something like `Either[Set[Throwable], Node]`. what do you think?